### PR TITLE
fix: update help_text for AAP credential type

### PR DESCRIPTION
**Pull request by hsong-rh**
https://issues.redhat.com/browse/AAP-31196

Update the help_text for RH AAP credential type.
Before:
image

After:
image

### CODE REVIEW

**Review by mkanoor — DISMISSED** [review]

File: src/aap_eda/core/migrations/0051_auto_20241011_1449.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+def update_credential_type_help_text(apps, help_text):
+    CredentialType = apps.get_model("core", "CredentialType")  # noqa: N806
+
+    for credential_type in CredentialType.objects.filter(
+        name=enums.DefaultCredentialType.AAP
+    ):
+        for field in credential_type.inputs["fields"]:
+            if field["id"] == "host":
+                field["help_text"] = help_text
```
Comment (Alex-Izquierdo) [inline src/aap_eda/core/migrations/0051_auto_20241011_1449.py:35]: Minor optimization: No need to continue the loop. 
```suggestion
                field["help_text"] = help_text
                break
```

File: src/aap_eda/core/migrations/0051_auto_20241011_1449.py
Comment (Alex-Izquierdo) [inline src/aap_eda/core/migrations/0051_auto_20241011_1449.py:1]: Can you rename the migration file with meaningful name?

File: src/aap_eda/core/migrations/0051_auto_20241011_1449.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Generated by Django 4.2.7 on 2024-10-11 14:49
```
Comment (Alex-Izquierdo) [inline src/aap_eda/core/migrations/0051_auto_20241011_1449.py:1]: This migration is not generated by django.

**Review by Alex-Izquierdo — COMMENTED** [review]
We should have a test that validates the data has been updated.

File: src/aap_eda/core/migrations/0051_auto_20241011_1449.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+def update_credential_type(apps, schema_editor):
+    update_credential_type_help_text(apps, new_help_text)
+
+
+def backward_credential_type(apps, schema_editor):
+    update_credential_type_help_text(apps, orig_help_text)
+
+
+def update_credential_type_help_text(apps, help_text):
```
Comment (Alex-Izquierdo) [inline src/aap_eda/core/migrations/0051_auto_20241011_1449.py:27]: We should have a test that validates the data has been updated.

File: src/aap_eda/core/migrations/0051_auto_20241011_1449.py (hunk 2 above)
Comment (hsong-rh) [reply src/aap_eda/core/migrations/0051_auto_20241011_1449.py]: I generated it using `makemigrations core --empty`

File: src/aap_eda/core/migrations/0051_auto_20241011_1449.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Generated by Django 4.2.7 on 2024-10-11 14:49
+
+from django.db import migrations
+
+from aap_eda.core import enums
+
+new_help_text = (
+    "Red Hat Ansible Automation Platform base URL to authenticate with.",
+    "For Event-Driven Ansible controller 2.5 with automation controller 2.4, use the following example: https://<<your_controller_host>>",  # noqa E501
```
[the remark this answers is not in the document]
Comment (hsong-rh) [reply src/aap_eda/core/migrations/0051_auto_20241011_1449.py]: @zkayyali812 The help_text comes from the mockup. Let me check with Tina

File: tests/unit/migrations/test_update_credential_type.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        "CONTROLLER_USERNAME": "{{username}}",
+        "CONTROLLER_PASSWORD": "{{password}}",
+        "CONTROLLER_VERIFY_SSL": "{{verify_ssl}}",
+        "CONTROLLER_OAUTH_TOKEN": "{{oauth_token}}",
+    }
+}
+
+
+@pytest.mark.django_db
+def test_migration():
```
Comment (Alex-Izquierdo) [inline tests/unit/migrations/test_update_credential_type.py:88]: I think this test can be shorter and much simpler, we can just apply the migration 0051 and validate the new value. 
Later we have to restore the migration as teardown. You can look at https://github.com/ansible/eda-server/blob/fea5fb162b9202a4f858dc4d0d69d92aaee7ac48/tests/unit/commands/test_wait_for_migrations.py#L23 as reference.

**Review by Alex-Izquierdo — APPROVED** [review]

File: tests/unit/migrations/test_update_credential_type.py (hunk 5 above)
Comment (hsong-rh) [reply tests/unit/migrations/test_update_credential_type.py]: changed

**Review by Alex-Izquierdo — APPROVED** [review]

**Review by mkanoor — APPROVED** [review]

**Pull request merged**